interop/relayer.go

```diff
@@ -93,42 +90,10 @@ func (r *L2ToL2MessageRelayer) Start(indexer *L2ToL2MessageIndexer, clients map[
 					close(sentMessageCh)
 					return nil
 				case sentMessage := <-sentMessageCh:
-					identifier := sentMessage.Identifier()
-					msg := sentMessage.Message()
-					calldata, err := msg.EventData()
-
-					if err != nil {
-						r.logger.Debug("failed to get event data", "err", err)
-						return fmt.Errorf("failed to get event data: %w", err)
-					}
-
-					// For some reason gas estimation does not work properly for some messages so estimating and padding gas (https://github.com/ethereum-optimism/supersim/issues/143)
-					txData, err := bindings.CrossL2InboxParsedABI.Pack("executeMessage", *identifier, predeploys.L2toL2CrossDomainMessengerAddr, calldata)
-					if err != nil {
-						r.logger.Debug("failed to create execute message calldata", "err", err)
-						return fmt.Errorf("failed to create execute message calldata: %w", err)
-					}
-					gasEstimate, err := client.EstimateGas(r.tasksCtx, ethereum.CallMsg{
-						From: transactor.From,
-						To:   &predeploys.CrossL2InboxAddr,
-						Data: txData,
-					})
-					if err != nil {
-						r.logger.Debug("failed to estimate gas for execute message tx", "err", err)
-						return fmt.Errorf("failed to estimate gas for execute message tx: %w", err)
-					}
-					// Pad gas by 33%.
-					paddedGas := (gasEstimate / 3) + gasEstimate
-					mu.Lock()
-					defer mu.Unlock()
-					originalTransactorGasLimit := transactor.GasLimit
-					transactor.GasLimit = paddedGas
-
-					if _, err := crossL2Inbox.ExecuteMessage(transactor, *identifier, predeploys.L2toL2CrossDomainMessengerAddr, calldata); err != nil {
+					if _, err := l2tol2CDM.RelayMessage(transactor, *sentMessage.Identifier(), sentMessage.message.Calldata); err != nil {
```

Comment:
How about we have a reference to the log or build the interop payload for the sent message event when handling it and store that within the `L2ToL2MessageStoreEntry` since that's the main use case for relaying the message?

And this module is responsible for the low-level details with the inbox

Reply:
sounds good, I added a `MessagePayload()`method to `L2ToL2MessageStoreEntry`